language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1305,6 +1304,45 @@ public static TypeSymbol getRawType(TypeSymbol typeDescriptor) {
         return typeDescriptor;
     }
 
+    /**
+     * Check if the provided union type is a union of members of provided type desc kind.
+     *
+     * @param typeSymbol Union type symbol
+     * @param typeDescKind    Type desc kind
+     * @return true if provided union contains members of provided type desc kind
+     */
+    public static boolean isUnionOfType(TypeSymbol typeSymbol, TypeDescKind typeDescKind) {
+        return typeSymbol.typeKind() == TypeDescKind.UNION &&
+                ((UnionTypeSymbol) typeSymbol).memberTypeDescriptors().stream()
+                        .map(CommonUtil::getRawType)
+                        .map(TypeSymbol::typeKind)
+                        .allMatch(kind -> kind == typeDescKind);
+    }
+
+    /**
+     * Given a type symbol, this method will get the record type symbols in the provided type. i.e. the provided type
+     * can be a record or a union of records for this to work.
+     *
+     * @param typeSymbol Type symbol from which record types need to be extracted.
+     * @return List of record type symbols, wrapped with a raw type and broader type container.
+     */
+    public static List<RawTypeSymbolWrapper<RecordTypeSymbol>> getRecordTypeSymbols(TypeSymbol typeSymbol) {
```

Comment:
Is it intended that we neglect the non-record type member types in the union types? If so shall we clearly mention that in the documentations?

Reply:
@nadeeshaan It wasn't my original idea. But I was refactoring the following method in CommonUtil. And it had considered only the record type members.
```
    protected List<Pair<TypeSymbol, TypeSymbol>> getRecordTypeDescs(BallerinaCompletionContext context,
                                                                    Node node) {
        ContextTypeResolver typeResolver = new ContextTypeResolver(context);
        Optional<TypeSymbol> resolvedType = node.apply(typeResolver);
        if (resolvedType.isEmpty()) {
            return Collections.emptyList();
        }
        TypeSymbol rawType = CommonUtil.getRawType(resolvedType.get());
        if (rawType.typeKind() == TypeDescKind.RECORD) {
            return Collections.singletonList(Pair.of(rawType, resolvedType.get()));
        }
        if (rawType.typeKind() == TypeDescKind.UNION) {
            return ((UnionTypeSymbol) rawType).memberTypeDescriptors().stream()
                    .filter(typeSymbol -> CommonUtil.getRawType(typeSymbol).typeKind() == TypeDescKind.RECORD)
                    .map(typeSymbol -> Pair.of(CommonUtil.getRawType(typeSymbol), typeSymbol))
                    .collect(Collectors.toList());
        }

        return Collections.emptyList();
    }
```
That's why I kept it this way. WDYT?

Reply:
For now, mentioned that this method will disregard non-record type members as a comment.